src/components/molecules/Visualizer/Engine/Cesium/hooks.ts

```diff
@@ -167,10 +178,60 @@ export default ({
   useEffect(() => {
     const viewer = cesium.current?.cesiumElement;
     if (!viewer || viewer.isDestroyed()) return;
-
     viewer.scene.requestRender();
   });
 
+  const handleDragLayer = useCallback(
+    (e: Entity, position: Cartesian3 | undefined, _context: Context): boolean | void => {
+      onDragLayer?.(e.id, convertCartesian3ToPosition(position));
+    },
+    [onDragLayer],
+  );
+
+  const handleDraggingLayer = useCallback(
+    (e: Entity, position: Cartesian3 | undefined, _context: Context): false | void => {
+      onDraggingLayer?.(e.id, convertCartesian3ToPosition(position));
+    },
+    [onDraggingLayer],
+  );
+
+  const handleDropLayer = useCallback(
+    (e: Entity, position: Cartesian3 | undefined, _context: Context): boolean | void => {
+      onDropLayer?.(e.id, convertCartesian3ToPosition(position));
+      if (cesium.current?.cesiumElement?.scene.screenSpaceCameraController) {
+        cesium.current.cesiumElement.scene.screenSpaceCameraController.enableRotate = true;
+      }
+    },
+    [onDropLayer],
+  );
+
+  const convertCartesian3ToPosition = (
+    pos?: Cartesian3,
+  ): { lat: number; lng: number; height: number } | undefined => {
+    if (!pos) return;
+    const cartographic =
+      cesium.current?.cesiumElement?.scene.globe.ellipsoid.cartesianToCartographic(pos);
+    if (!cartographic) return;
+    return {
+      lat: CesiumMath.toDegrees(cartographic.latitude),
+      lng: CesiumMath.toDegrees(cartographic.longitude),
+      height: cartographic.height,
+    };
+  };
+
+  //Enable Drag and Drop Layers
+  const DELAY = 1000;
```

Comment:
Can you play with this a little more or chat with the team? For me, 1 second feels a little long.

Reply:
yesterday we set to less than 0.01s I guess, it will course some error when we select the marker, I think for now 1 s is ok. This needs more tests I think.